syncer/sharding-meta/shardmeta.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shardmeta
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/siddontang/go-mysql/mysql"
+
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/utils"
+)
+
+const (
+	// MetaTableFormat is used in meta table name constructor
+	MetaTableFormat = "%s_syncer_sharding_meta"
+)
+
+// DDLItem records ddl information used in sharding sequence organization
+type DDLItem struct {
+	FirstPos mysql.Position `json:"first-pos"` // first DDL's binlog Pos, not the End_log_pos of the event
+	DDLs     []string       `json:"ddls"`      // DDLs, these ddls are in the same QueryEvent
+	Source   string         `json:"source"`    // source table ID
+}
+
+// String returns the item's format string value
+func (item *DDLItem) String() string {
+	return fmt.Sprintf("first-pos: %s ddls: %+v source: %s", item.FirstPos, item.DDLs, item.Source)
+}
+
+// ShardingSequence records a list of DDLItem
+type ShardingSequence struct {
+	Items []*DDLItem `json:"items"`
+}
+
+// ShardingMeta stores sharding ddl sequence
+// including global sequence and each source's own sequence
+type ShardingMeta struct {
```

Comment:
move this block of code before `NewShardingMeta` and move `NewDDLItem` after `type DDLItem struct`?

so methods for `DDLItem`/`ShardingSequence`/`ShardingMeta` close to the definition of them.